gradle-jdks-settings/src/main/java/com/palantir/gradle/jdks/settings/PatchToolchainJdkPlugin.java

```diff
@@ -0,0 +1,203 @@
+            // src/main/java/org/gradle/api/internal/properties/GradleProperties.java#L28
+            if (method.getName().equals("find")) {
+                if (args.length == 1 && args[0].equals("org.gradle.java.installations.auto-detect")) {
+                    return "false";
+                }
+                if (args.length == 1 && args[0].equals("org.gradle.java.installations.auto-download")) {
+                    return "false";
+                }
+                if (args.length == 1 && args[0].equals("org.gradle.java.installations.paths")) {
+                    return localToolchains.stream()
+                            .map(Path::toAbsolutePath)
+                            .map(Path::toString)
+                            .collect(Collectors.joining(","));
+                }
+            }
+            try {
+                return GradleProperties.class
+                        .getDeclaredMethod(method.getName(), method.getParameterTypes())
+                        .invoke(originalGradleProperties, args);
+            } catch (InvocationTargetException e) {
+                throw e.getCause();
+            }
+        }
+
+        private static List<Path> getInstalledToolchains(Path gradleJdksLocalDirectory) {
+            Path installationDirectory = getToolchainInstallationDir();
+            Os os = CurrentOs.get();
+            Arch arch = CurrentArch.get();
+            try (Stream<Path> stream = Files.list(gradleJdksLocalDirectory).filter(Files::isDirectory)) {
+                return stream.map(path -> path.resolve(String.format("%s/%s/local-path", os, arch)))
+                        .map(path -> getToolchain(path, installationDirectory))
+                        .flatMap(Optional::stream)
+                        .collect(Collectors.toList());
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to list the local installation paths", e);
+            }
+        }
+
+        private static Optional<Path> getToolchain(Path gradleJdkConfigurationPath, Path installationDirectory) {
+            try {
+                String localFilename =
+                        Files.readString(gradleJdkConfigurationPath).trim();
+                Path installationPath = installationDirectory.resolve(localFilename);
+                if (!Files.exists(installationPath)) {
+                    logger.warn("Could not find toolchain at {}", installationPath);
+                    return Optional.empty();
+                }
+                return Optional.of(installationPath);
+            } catch (IOException e) {
+                throw new RuntimeException(
+                        String.format("Failed to get the toolchain configured at path=%s", gradleJdkConfigurationPath),
+                        e);
+            }
+        }
+    }
+
+    @Override
+    public void apply(Settings settings) {
```

Comment:
Would put the `apply` first in the class, reading it just flows a bit better this way.